pkg/spoke/addon/configuration.go

```diff
@@ -0,0 +1,51 @@
+package addon
+
+import (
+	"fmt"
+
+	addonv1alpha1 "github.com/open-cluster-management/api/addon/v1alpha1"
+)
+
+const (
+	installationNamespaceAnnotation = "installNamespace"
+	bootstrapSecretAnnotation       = "bootstrapSecret"
+	registrationAnnotation          = "registrations"
+)
+
+// certSubject contains fields which are used to generate csr subject and can be customized by users.
+type certSubject struct {
+	CommonName         string
+	Organization       []string
+	OrganizationalUnit []string
+}
+
+// registrationConfig contains necessary information for addon registration.
+type registrationConfig struct {
+	SignerName string
+	Subject    certSubject
+}
+
+// addonConfig contains addon configuration information.
+type addOnConfig struct {
+	AddOnName             string
+	InstallationNamespace string
+	BootstrapSecretName   string
+	Registrations         []registrationConfig
```

Comment:
dead?

Reply:
this is for addon registration part, we will use this in that part

Reply:
remove this now